### PR TITLE
HW9 Task02

### PR DESCRIPTION
**Pull request by LinaKrevetochka**

### CODE REVIEW

File: hw/hw_09_task_02.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# flake8: noqa: R503
+class SuppressorCls:
+    def __init__(self, exception):
+        self.exception = exception
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if exc_type == self.exception:
```
Comment (Cjkjvfnby) [inline hw/hw_09_task_02.py]: Can we inline it?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Do both versions work in the same way?

**Comment by LinaKrevetochka** [comment]
> Do both versions work in the same way?

I don't really understand you. It's different structures, so they can work in different ways, can't they? They have the same behavior in tests.

File: test/test_02_hw_09.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from hw.hw_09_task_02 import SuppressorCls, suppressor_gen
+
+import pytest
+
+
+def test_class_suppressed_passed_exception():
+    with SuppressorCls(IndexError):
+        [][2]
```
Comment (Cjkjvfnby) [inline test/test_02_hw_09.py:8]: What does this code do?

It raises IndexError.

Can we read Python code as an English phrase?  Yes! 
 
```python
raise IndexError()
```

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Please add a test for both cases:

```
with ... (Exception):
   raise IndexError()
```

**Comment by LinaKrevetochka** [comment]
Ok, I got it. "Ecxept"  also catches all subclasses. We should also check it in class.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**